File: Python/underautomation/universal_robots/rtde/rtde_input_values.py
import typing
from underautomation.universal_robots.rtde.rtde_input_setup_item import RtdeInputSetupItem
from underautomation.universal_robots.rtde.rtde_input_data import RtdeInputData
from underautomation.universal_robots.rtde.rtde_bit_registers_value import RtdeBitRegistersValue
from underautomation.universal_robots.rtde.rtde_int_registers_value import RtdeIntRegistersValue
from underautomation.universal_robots.rtde.rtde_double_registers_value import RtdeDoubleRegistersValue
from underautomation.universal_robots.common.cartesian_coordinates import CartesianCoordinates
from underautomation.universal_robots.rtde.rtde_base_values_1 import RtdeBaseValues1
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Rtde import RtdeInputValues as rtde_input_values

class RtdeInputValues(RtdeBaseValues1[RtdeInputData]):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = rtde_input_values()
		else:
			self._instance = _internal
	def set_value(self, data: RtdeInputData, index: int, value: typing.Any) -> None:
		self._instance.SetValue(data._instance, index, value)
	def get_value(self, item: RtdeInputSetupItem) -> typing.Any:
		return self._instance.GetValue(item._instance)
	def reset(self) -> None:
		self._instance.Reset()
	@property
	def speed_slider_mask(self) -> int:
		return self._instance.SpeedSliderMask
	@speed_slider_mask.setter
	def speed_slider_mask(self, value: int):
		self._instance.SpeedSliderMask = value
	@property
	def speed_slider_fraction(self) -> float:
		return self._instance.SpeedSliderFraction
	@speed_slider_fraction.setter
	def speed_slider_fraction(self, value: float):
		self._instance.SpeedSliderFraction = value
	@property
	def standard_digital_output_mask(self) -> int:
		return self._instance.StandardDigitalOutputMask
	@standard_digital_output_mask.setter
	def standard_digital_output_mask(self, value: int):
		self._instance.StandardDigitalOutputMask = value
	@property
	def configurable_digital_output_mask(self) -> int:
		return self._instance.ConfigurableDigitalOutputMask
	@configurable_digital_output_mask.setter
	def configurable_digital_output_mask(self, value: int):
		self._instance.ConfigurableDigitalOutputMask = value
	@property
	def standard_digital_output(self) -> int:
		return self._instance.StandardDigitalOutput
	@standard_digital_output.setter
	def standard_digital_output(self, value: int):
		self._instance.StandardDigitalOutput = value
	@property
	def configurable_digital_output(self) -> int:
		return self._instance.ConfigurableDigitalOutput
	@configurable_digital_output.setter
	def configurable_digital_output(self, value: int):
		self._instance.ConfigurableDigitalOutput = value
	@property
	def standard_analog_output_mask(self) -> int:
		return self._instance.StandardAnalogOutputMask
	@standard_analog_output_mask.setter
	def standard_analog_output_mask(self, value: int):
		self._instance.StandardAnalogOutputMask = value
	@property
	def standard_analog_output_type(self) -> int:
		return self._instance.StandardAnalogOutputType
	@standard_analog_output_type.setter
	def standard_analog_output_type(self, value: int):
		self._instance.StandardAnalogOutputType = value
	@property
	def standard_analog_output0(self) -> float:
		return self._instance.StandardAnalogOutput0
	@standard_analog_output0.setter
	def standard_analog_output0(self, value: float):
		self._instance.StandardAnalogOutput0 = value
	@property
	def standard_analog_output1(self) -> float:
		return self._instance.StandardAnalogOutput1
	@standard_analog_output1.setter
	def standard_analog_output1(self, value: float):
		self._instance.StandardAnalogOutput1 = value
	@property
	def input_bt_registers0_to31(self) -> int:
		return self._instance.InputBtRegisters0To31
	@input_bt_registers0_to31.setter
	def input_bt_registers0_to31(self, value: int):
		self._instance.InputBtRegisters0To31 = value
	@property
	def input_bt_registers32_to63(self) -> int:
		return self._instance.InputBtRegisters32To63
	@input_bt_registers32_to63.setter
	def input_bt_registers32_to63(self, value: int):
		self._instance.InputBtRegisters32To63 = value
	@property
	def input_bit_registers(self) -> RtdeBitRegistersValue:
		return RtdeBitRegistersValue(self._instance.InputBitRegisters)
	@property
	def input_int_registers(self) -> RtdeIntRegistersValue:
		return RtdeIntRegistersValue(self._instance.InputIntRegisters)
	@property
	def input_double_registers(self) -> RtdeDoubleRegistersValue:
		return RtdeDoubleRegistersValue(self._instance.InputDoubleRegisters)
	@property
	def external_force_torque(self) -> CartesianCoordinates:
		return CartesianCoordinates(None, None, None, None, None, None, self._instance.ExternalForceTorque)
	@external_force_torque.setter
	def external_force_torque(self, value: CartesianCoordinates):
		self._instance.ExternalForceTorque = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Rtde/RtdeInputValues.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace Rtde {

	public class RtdeInputValues : RtdeBaseValues<RtdeInputData> {


		public RtdeInputValues()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}


		protected override RtdeValue[] InternalValues { get; }


		protected override RtdeValue InternaleGetValue(RtdeInputData data)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public void SetValue(RtdeInputSetupItem item, object value)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}


		public void SetValue(RtdeInputData data, object value)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}


		public void SetValue(RtdeInputData data, int index, object value)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}


		public object GetValue(RtdeInputSetupItem item)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public void Reset()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// 0 = don't change speed slider with this input, 1 = use speed_slider_fraction to set speed slider value
		/// </summary>
		public uint SpeedSliderMask { get; set; }

		/// <summary>
		/// new speed slider value
		/// </summary>
		public double SpeedSliderFraction { get; set; }

		/// <summary>
		/// Standard digital output bit mask
		/// </summary>
		public byte StandardDigitalOutputMask { get; set; }

		/// <summary>
		/// Configurable digital output bit mask
		/// </summary>
		public byte ConfigurableDigitalOutputMask { get; set; }

		/// <summary>
		/// Standard digital outputs
		/// </summary>
		public byte StandardDigitalOutput { get; set; }

		/// <summary>
		/// Configurable digital outputs
		/// </summary>
		public byte ConfigurableDigitalOutput { get; set; }

		/// <summary>
		/// Standard analog output mask
		/// </summary>
		public byte StandardAnalogOutputMask { get; set; }

		/// <summary>
		/// Output domain {0=current[mA], 1=voltage[V]}. Bits 0-1: standard_analog_output_0 | standard_analog_output_1
		/// </summary>
		public byte StandardAnalogOutputType { get; set; }

		/// <summary>
		/// Standard analog output 0 (ratio) [0..1]
		/// </summary>
		public double StandardAnalogOutput0 { get; set; }

		/// <summary>
		/// Standard analog output 1 (ratio) [0..1]
		/// </summary>
		public double StandardAnalogOutput1 { get; set; }

		/// <summary>
		/// General purpose bits. This range of the boolean input registers is reserved for FieldBus/PLC interface usage.
		/// </summary>
		public uint InputBtRegisters0To31 { get; set; }

		/// <summary>
		/// General purpose bits. This range of the boolean input registers is reserved for FieldBus/PLC interface usage.
		/// </summary>
		public uint InputBtRegisters32To63 { get; set; }

		/// <summary>
		/// 64 general purpose bits. X: [64..127] - The upper range of the boolean input registers can be used by external RTDE clients (i.e URCAPS).
		/// </summary>
		public RtdeBitRegistersValue InputBitRegisters { get; }

		/// <summary>
		/// 48 general purpose integer registers. X: [0..23] - The lower range of the integer input registers is reserved for FieldBus/PLC interface usage. X: [24..47] - The upper range of the integer input registers can be used by external RTDE clients (i.e URCAPS).
		/// </summary>
		public RtdeIntRegistersValue InputIntRegisters { get; }

		/// <summary>
		/// 48 general purpose double registers. X: [0..23] - The lower range of the double input registers is reserved for FieldBus/PLC interface usage. X: [24..47] - The upper range of the double input registers can be used by external RTDE clients (i.e URCAPS).
		/// </summary>
		public RtdeDoubleRegistersValue InputDoubleRegisters { get; }

		/// <summary>
		/// Input external wrench when using ft_rtde_input_enable builtin.
		/// </summary>
		public CartesianCoordinates ExternalForceTorque { get; set; }
	}
}
